cartographer/io/file_writer.h

```diff
@@ -45,21 +45,31 @@ class FileWriter {
   virtual std::string GetFilename() = 0;
 };
 
-// An Implementation of file using std::ofstream.
-class StreamFileWriter : public FileWriter {
+class StreamWriter : public FileWriter {
  public:
-  ~StreamFileWriter() override;
+  virtual ~StreamWriter() override;
 
-  StreamFileWriter(const std::string& filename);
+  StreamWriter(std::unique_ptr<std::ostream> ostream,
+               const std::string& filename);
 
-  bool Write(const char* data, size_t len) override;
-  bool WriteHeader(const char* data, size_t len) override;
-  bool Close() override;
+  virtual bool Write(const char* data, size_t len) override;
+  virtual bool WriteHeader(const char* data, size_t len) override;
+  virtual bool Close() override;
   std::string GetFilename() override;
 
- private:
+ protected:
```

Comment:
data members have to be public.
https://google.github.io/styleguide/cppguide.html#Access_Control